blocks/identity-block/components/FormInputField/index.jsx

```diff
@@ -0,0 +1,94 @@
+import React, { useState } from 'react';
+
+import { PrimaryFont } from '@wpmedia/shared-styles';
+
+import './styles.scss';
+
+export const FIELD_TYPES = {
+  EMAIL: 'email',
+  TEXT: 'text',
+  PASSWORD: 'password',
+};
+
+const FormInputField = ({
+  defaultValue = '',
+  label,
+  name,
+  isDefaultError = false,
+  onChange = () => {},
+  placeholder,
+  minLength = 0,
+  maxLength = 0,
+  required = false,
+  tip,
+  type = FIELD_TYPES.TEXT,
+  validationErrorMessage,
+  validationPattern,
+}) => {
+  const [error, setError] = useState(isDefaultError);
+  const [errorMessage, setErrorMessage] = useState(validationErrorMessage);
+
+  const handleBlur = (event) => {
+    setError(!!event.target.validationMessage);
+    setErrorMessage(event.target.validationMessage
+      ? validationErrorMessage || event.target.validationMessage
+      : '');
+  };
+
+  const handleChange = (event) => {
+    if (error) {
+      setError(!!event.target.validationMessage);
+      setErrorMessage(event.target.validationMessage
+        ? validationErrorMessage || event.target.validationMessage
+        : '');
+    }
+    onChange(event);
```

Comment:
Should we control what is passed back here? Should we only pass back the value of the input?

Reply:
Debated this... I considered only sending back the values, but I didn't know what we would need for future support.  I can definitely do that and come back later to add the additional needs later, but that might involve editing other forms that also use this field.  Figured it was safe to pass back the event object since it's a synthetic field and emulating the field onChange was the goal.

Reply:
Makes sense - the way to make this future proof would be to send back an object like ```{ value: value}``` and that way we would be able to expand on the keys being passed back without breaking existing uses

Reply:
Updated to do this instead of passing the event.